etc/bash_completion.d/brew

```diff
@@ -47,21 +47,20 @@ __brew_complete_formulae ()
     local cur="${COMP_WORDS[COMP_CWORD]}"
     local lib=$(brew --repository)/Library
     local taps=${lib}/Taps
-    local ff=$(\ls ${lib}/Formula 2>/dev/null | sed 's/\.rb//g')
-    local af=$(\ls ${lib}/Aliases 2>/dev/null)
-    local tf file
-
-    for file in ${taps}/*/*/*.rb ${taps}/*/*/Formula/*.rb ${taps}/*/*/HomebrewFormula/*.rb; do
-        [ -f "$file" ] || continue
-        file=${file/"Formula/"/}
-        file=${file/"HomebrewFormula/"/}
-        file=${file#${lib}/Taps/}
-        file=${file%.rb}
-        file=${file/homebrew-/}
-        tf="${tf} ${file}"
-    done
+    
+    # capture tap/Formula/foo.rb, tap/HomebrewFormula/foo.rb
+    local ff=$(find $taps -depth 4 -path '*Formula/*rb' | sed -E "s|$taps/([^/]+)/homebrew-([^/]+)/(Homebrew)?Formula/(.*)\.rb|\1-\2/\4|")
```

Comment:
Seems like you could just run `find` once and strip the results each time you need/don't. It's also worth noting `sed -E` will behave differently on Linux, I believe (if you care)